### PR TITLE
[REVIEW] Row-wise op warning when necessary

### LINKED ISSUE #6018: [BUG] DataFrame.sum() with axis=1 warning message should be displayed only when not supported datatype is used
**Describe the bug**
Hi! When using `DataFrame.sum(axis=1)`, the following warning message is displayed:

```
/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py:5367: UserWarning: Row-wise operations currently only support int, float, and bool dtypes.
  "Row-wise operations currently only support int, float, "
```
In the picture below, it can be seen that `sum()` is being calculated on int64 columns only. Therefore, I am not sure if the warning message should be displayed.

image

**Steps/Code to reproduce bug**
Create a DataFrame with int64 columns only, and try to sum their values row by row.

**Expected behavior**
No warning message is displayed when int or float columns are used.

**Environment overview (please complete the following information)**
RAPIDS 0.15 nightly build.

### PR DESCRIPTION
**Pull request by isVoid**
Closes #6018 

- Guards the warning with column counts. Add "ignore non-numeric columns" to the warnings.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -5359,6 +5355,12 @@ def _prepare_for_rowwise_op(self):
 
         filtered = self.select_dtypes(include=[np.number, np.bool])
         common_dtype = np.find_common_type(filtered.dtypes, [])
+        if len(filtered.columns) < len(self.columns):
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:5358]: Unfortunately `DataFrame.columns` is somewhat expensive to compute. Let's use `df._num_columns` instead.

Comment (shwina) [reply python/cudf/cudf/core/dataframe.py]: Ditto for `self` :)

Comment (isVoid) [reply python/cudf/cudf/core/dataframe.py]: Yup!

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@kkraus14 this is good to go; could you please admin merge it when you have a chance?

**Pull request merged**